### PR TITLE
Added support for smart_rollup_add_messages op and relevant tests

### LINKED ISSUE #2309: [Mumbai] Add support in Contract API for Smart_rollup_add_messages
Add support for the new operation `smart_rollup_add_messages` in these following providers:
- RpcContractProvider
- EstimateProvider
- PrepareProvider

Additional requirements:
- Update types/interfaces as necessary
- Add unit and integration tests
- Update documentation

example operation result:
https://mumbainet.ecadinfra.com/chains/main/blocks/BLytaxnQyL7B4XASV6PCCc55oBNvWfZAcALgGU5cHv6dcCuk8rK

opHash: `op3GFxBQArsgC3eHjEiw4Qp31jHrprKfftcgwibKAwnANpzWncG`

```
[
  {
    "protocol": "PtMumbaiiFFEGbew1rRjzSPyzRbA51Tm3RVZL5suHPxSZYDhCEc",
    "chain_id": "NetXQw6nWSnrJ5t",
    "hash": "op3GFxBQArsgC3eHjEiw4Qp31jHrprKfftcgwibKAwnANpzWncG",
    "branch": "BL7USLDrUeuMzDUX6PdxuCbhngMYCcnmy9WFuotiVRmdBftZeDv",
    "contents": [
      {
        "kind": "smart_rollup_add_messages",
        "source": "tz2Q3yRaczTqZVf3ZQvwiiTqKjhJFyDzeRSz",
        "fee": "398",
        "counter": "12191",
        "gas_limit": "1103",
        "storage_limit": "0",
        "message": [
          "0000000031010000000b48656c6c6f20776f726c6401cc9e352a850d7475bf9b6cf103aa17ca404bc9dd000000000764656661756c74"
        ],
        "metadata": {
          "balance_updates": [
            {
              "kind": "contract",
              "contract": "tz2Q3yRaczTqZVf3ZQvwiiTqKjhJFyDzeRSz",
              "change": "-398",
              "origin": "block"
            },
            {
              "kind": "accumulator",
              "category": "block fees",
              "change": "398",
              "origin": "block"
            }
          ],
          "operation_result": {
            "status": "applied",
            "consumed_milligas": "1002777"
          }
        }
      }
    ],
    "signature": "sigSUjvKxjAZ4dBWbo4idKKwFDVfLtYscMMqHoQY8KgyghtyaswECPaBhjK921vj2uEsdKD7WJTeVVT1ZDcvwp8rkRuEW9kv"
  }
]
```

### PR DESCRIPTION
**Pull request by dsawali**
Added smart_rollup_add_messages op support in the 
- estimate provider
- contract provider
- prepare provider

Thank you for your contribution to Taquito.

Before submitting this PR, please make sure:
closes #2309 

- [x] Your code builds cleanly without any errors or warnings
- [x] You have run the linter against the changes
- [x] You have added unit tests (if relevant/appropriate)
- [x] You have added integration tests (if relevant/appropriate)
- [x] All public methods or types have TypeDoc coverage with a complete description, and ideally an @example
- [x] You have added or updated corresponding documentation
- [ ] If relevant, you have written a first draft summary describing the change for inclusion in Release Notes. 

In this PR, please also make sure: 

- [x] You have linked this PR to the issue by putting `closes #TICKETNUMBER` in the description box (when applicable)
- [x] You have added a concise description on your changes
## Release Note Draft Snippet

__If relevant, please write a summary of your change that will be suitable for
inclusion in the Release Notes for the next Taquito release.__

### CODE REVIEW

File: packages/taquito/test/contract/rpc-contract-provider.spec.ts [hunk 1]
```diff
@@ -1918,6 +1920,83 @@ describe('RpcContractProvider test', () => {
...
+      mockEstimate.reveal.mockResolvedValue(undefined);
+      const estimate = new Estimate(1230000, 93, 142, 250);
+      mockEstimate.smartRollupAddMessages.mockResolvedValue(estimate);
+      const op = await rpcContractProvider.smartRollupAddMessages({
+        message: [
+          '0000000031010000000b48656c6c6f20776f726c6401cc9e352a850d7475bf9b6cf103aa17ca404bc9dd000000000764656661756c74',
+        ],
+      });
+
+      console.log(JSON.stringify(op.raw));
```
Comment (zainen) [inline packages/taquito/test/contract/rpc-contract-provider.spec.ts:1935]: remove?

File: packages/taquito/test/prepare/prepare-provider.spec.ts [hunk 2]
```diff
@@ -889,6 +890,80 @@ describe('PrepareProvider test', () => {
...
+
+    it('should be able to prepare smartRollupAddMessages op without reveal when estimate is undefined', async () => {
+      jest.spyOn(context.estimate, 'reveal').mockResolvedValue(undefined);
+
+      const prepared = await prepareProvider.smartRollupAddMessages({
+        message: [
+          '0000000062010000000b48656c6c6f20776f726c6401bdb6f61e4f12c952f807ae7d3341af5367887dac000000000764656661756c74010000000b48656c6c6f20776f726c6401bdb6f61e4f12c952f807ae7d3341af5367887dac000000000764656661756c74',
+        ],
+      });
+      console.log(JSON.stringify(prepared));
```
Comment (zainen) [inline packages/taquito/test/prepare/prepare-provider.spec.ts:942]: same

File: docs/smart_rollups.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
+---
+title: Smart Rollups
+author: Davis Sawali 
+---
+
+# Smart Optimistic Rollups
+
+Rollups are a permissionless scaling for the Tezos blockchain. The idea is that anyone can originate and operate one or more rollups, increasing the overall throughput of the Tezos blockchain.
```
Comment (zainen) [inline docs/smart_rollups.md:8]: sounds kind of awkward 
`Rollups are a permissionless scaling for` maybe `Rollups are a permissionless scaling implementation for`

**Review by zainen — APPROVED** [review]
nice job! 
you didnt run into any estimate problems like i did?

File: docs/smart_rollups.md (hunk 3 above)
Comment (dsawali) [reply docs/smart_rollups.md:8]: good catch, your suggestion sounds a lot better

File: packages/taquito/test/contract/rpc-contract-provider.spec.ts (hunk 1 above)
Comment (dsawali) [reply packages/taquito/test/contract/rpc-contract-provider.spec.ts:1935]: whoops good catch

File: packages/taquito/test/prepare/prepare-provider.spec.ts (hunk 2 above)
Comment (dsawali) [reply packages/taquito/test/prepare/prepare-provider.spec.ts:942]: > nice job! you didnt run into any estimate problems like i did?

not really, no. Aside from the gasLimit requirement I mentioned in the docs, everything seemed okay.

File: integration-tests/contract-batch.spec.ts [hunk 4]
```diff
@@ -4,215 +4,215 @@ import { managerCode } from './data/manager_code';
 import { MANAGER_LAMBDA, OpKind } from '@taquito/taquito';
 
 CONFIGS().forEach(({ lib, rpc, setup, knownBaker, knownContract, createAddress }) => {
-    const Tezos = lib;
```
Comment (hui-an-yang) [inline integration-tests/contract-batch.spec.ts:7]: guess that it's only format changes in this file, right?

File: docs/smart_rollups.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
+---
+title: Smart Rollups
+author: Davis Sawali 
+---
+
+# Smart Optimistic Rollups
+
+Rollups are a permissionless scaling implementation for the Tezos blockchain. The idea is that anyone can originate and operate one or more rollups, increasing the overall throughput of the Tezos blockchain.
+
+In Taquito, we have implemented some of the operations included in Mumbai protocol update in regards to smart rollups. In this document, we will go through the ones that we support. We will also go over smart rollups in a more high level view, if you'd like to understand the feature a bit deeper, you can refer to [this document](https://tezos.gitlab.io/mumbai/smart_rollups.html).
```
Comment (zainen) [inline docs/smart_rollups.md:10]: what do you think about? 
`We will also go over smart rollups for a high level overview,`

**Review by zainen — APPROVED** [review]
looks good to me 
how formal do we want to be for our documentation?

File: packages/taquito/src/prepare/prepare-provider.ts [hunk 6]
```diff
@@ -219,7 +221,12 @@ export class PrepareProvider implements PreparationProvider {
             ...op,
             period: currentVotingPeriod?.voting_period.index,
           };
-
+        case OpKind.SMART_ROLLUP_ADD_MESSAGES:
```
Comment (hui-an-yang) [inline packages/taquito/src/prepare/prepare-provider.ts:224]: Seems to return the same thing as line 189-193, would be great to add a line of `case OpKind.SMART_ROLLUP_ADD_MESSAGES:` below line 188 to reduce repetition

File: packages/taquito/src/contract/rpc-contract-provider.ts [hunk 7]
```diff
@@ -697,6 +700,37 @@ export class RpcContractProvider
     );
   }
 
+  /**
+   * @description
```
Comment (hui-an-yang) [inline packages/taquito/src/contract/rpc-contract-provider.ts:704]: description missing

File: docs/smart_rollups.md (hunk 5 above)
Comment (dsawali) [reply docs/smart_rollups.md:10]: I don't think 'for' sounds right in this context. I think we can do 'We will also go over smart rollups on a high level,...'

Comment (dsawali) [reply docs/smart_rollups.md]: I changed it to something a bit more verbose and less vague

File: docs/smart_rollups.md [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+const op = await Tezos.contract.smartRollupAddMessages({
+    message: [
+        '0000000031010000000b48656c6c6f20776f726c6401bdb6f61e4f12c952f807ae7d3341af5367887dac000000000764656661756c74'
+    ]
+});
+
+await op.confirmation();
+```
+
+- `message` property is an encoded outbox message. For more information on how to encode or what message gets encoded, refer to [this document](https://tezos.gitlab.io/mumbai/smart_rollups.html#sending-an-external-inbox-message)
```
Comment (zainen) [inline docs/smart_rollups.md:42]: just a thought
would it be ideal to specify that multiple of these encoded messages can be added the the message: array?

Comment (zainen) [reply docs/smart_rollups.md:42]: there is an example that i made in the local forger that shows when multiple encoded messages are added

**Review by zainen — APPROVED** [review]

File: docs/smart_rollups.md (hunk 8 above)
Comment (zainen) [reply docs/smart_rollups.md:42]: but is very minor and not sure when it would be used

Comment (dsawali) [reply docs/smart_rollups.md:42]: yeah I can add a bit more detail clarifying that

Comment (dsawali) [reply docs/smart_rollups.md:42]: though it should be pretty clear since the message property param is required to be an array

**Pull request merged**